### PR TITLE
SQL-1116: Implement evergreen build file with initial connector files

### PR DESCRIPTION
**Pull request by bucaojit**
Added initial connector files in the `connector/` directory that are mostly unchanged from the sample generator.  
Tested the built `MongoDBAtlasODBC.pqx` connector file with PowerBI, it shows up in the `Get Data` dialogue and shows the `Hello from MongoDBAtlasODBC...` message.
Added unit test as a placeholder, it runs the unit test but does not currently check that the output is correct.  Signing task is also a placeholder.

### CODE REVIEW

**Review by mattChiaravalloti — APPROVED** [review]
Looks great 💯 great fresh start for another project!

File: evergreen.yml [hunk 1]
```diff
@@ -33,51 +65,144 @@ functions:
...
-  "build connector":
+  "install power query tools":
     - command: shell.exec
       params:
         shell: bash
         working_dir: mongo-powerbi-connector
         script: |
-          echo 'PLACEHOLDER' > mongodb-powerbi.pqx
+          ${prepare_shell}
+          curl  -L -o tools.zip https://www.nuget.org/api/v2/package/Microsoft.PowerQuery.SdkTools/2.111.5
```
Comment (nbagnard) [inline evergreen.yml:92]: Question: Does the SDK version change often? What are we planning to do? Can we stay on an older version of the SDK for as long as we want or does PowerBI require your connector to use a specific version to let you run it?

**Review by nbagnard — APPROVED** [review]
The only thing that I'd like you to check is if we could use Mongo icons instead of the generic Windows ones. Depending on where this shows up, you might want the MongoDB logo or a "connector" logo. You can take a look in the Branding Asset folder.
I've done a quick search and one of these for example could match (there is a button to download a png with the requested size):
https://brandfolder.com/mongodb/all-branded-assets#!asset/bq8sjr3nnmf6sgczx98r82q
https://brandfolder.com/mongodb/all-branded-assets#!asset/295k846pm74t2fnpbq4bpknr
https://brandfolder.com/mongodb/all-branded-assets#!asset/qmsnkwrjpxsckgj3qm9spb

File: evergreen.yml (hunk 1 above)
Comment (bucaojit) [reply evergreen.yml:92]: They seem to be updating the versions often, about 5 versions in the last two months [here](https://www.nuget.org/packages/Microsoft.PowerQuery.SdkTools#versions-body-tab).  I tested without specifying the `2.111.5` version and it was able to download the package, I'll make that change.

**Comment by bucaojit** [comment]
@nbagnard I was looking into the icons as well.  I have a set of icons that I've scaled to the different sizes that I'll add to my SQL-1119 change based on the leaf icons here: https://www.mongodb.design/component/logo/example/

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**